File: src/com/dafttech/terra/graphics/gui/containers/ContainerOnscreen.java
package com.dafttech.terra.graphics.gui.containers;

import com.badlogic.gdx.Gdx;
import com.dafttech.terra.game.world.Vector2;

public class ContainerOnscreen extends GUIContainer {
    public ContainerOnscreen(Vector2 p) {
        super(p, new Vector2(Gdx.graphics.getWidth(), Gdx.graphics.getHeight()));
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/containers/ContainerOnscreen.scala
package com.dafttech.terra.engine.gui.containers

import com.badlogic.gdx.Gdx
import com.dafttech.terra.engine.vector.Vector2d

class ContainerOnscreen extends GUIContainer(p = Vector2d.Zero, s = Vector2d(Gdx.graphics.getWidth, Gdx.graphics.getHeight)) {
  private[containers] var active: Boolean = false

  def setActive(b: Boolean): Unit = active = b

  override def providesActiveHierarchy: Boolean = active
}